ProjectHub2/services/facebookService.js

```diff
@@ -24,90 +19,127 @@ function makeApiCall(oauthOptions, callback, hashtag) {
+    object.type = 'photo';
+    
+    return object;
 }
 
-function setPagePostsToDb(data) {
-    MongoClient.connect(url, function (err, db) {
-        var collection = db.collection(collectionName);
-        var object = {};
-        for (var i = 0; i < data['albums']['data'].length; i++) // albums
-        {
-        for (var j = 0; j < data['albums']['data'][i]['photos']['data'].length; j++) { // items in albums
-            object['_id'] =  'fb' + data['albums']['data'][i]['photos']['data'][j]['id'];
-            var date = new Date(data['albums']['data'][i]['photos']['data'][j]['created_time']);
-            object['date'] = date;
-            object['from'] = data['albums']['data'][i]['photos']['data'][j]['from'];
-                if (data['albums']['data'][i]['photos']['data'][j].hasOwnProperty('name'))
-                    object['description'] = data['albums']['data'][i]['photos']['data'][j]['name'];
-                else
-                    object['description'] = '';
-                
-                object['url'] = data['albums']['data'][i]['photos']['data'][j]['images'][0]['source'];
-                object['likes'] = data['albums']['data'][i]['photos']['data'][j]['likes']['summary']['total_count'];
-                object['comments'] = data['albums']['data'][i]['photos']['data'][j]['comments']['summary']['total_count'];
-                collection.save(object);
-            }
-        }
-    db.close(); 
-    }); 
+function makePhotoArray(data) {
+    var array = [];
+    var object = {};
+    data.albums.data.forEach(function(album) {
+        console.log('album');
+        album.photos.data.forEach(function(item) {
+            object = makePhotoObject(item);
+            array.push(object);
+            });
+    });
+    console.log(array);
+    return array;
 }
 
-getPagePosts(myOptions, setPagePostsToDb)
+// getPhotoPosts(myOptions, makeMediaArray);
+
+// ~~~~~~~~~~~~~~~~~~~~~~ videos ~~~~~~~~~~~~~~~~~~~~~~~
+
+function getVideoPosts(options) {
+    var oauthOptions = {
+        url: 'https://graph.facebook.com/v2.6/' + options['pageId'] + '?fields=videos{created_time,description,from,id,likes.limit(1).summary(true),comments.limit(1).summary(true),source}&access_token=' + options['appId'] + '|' + options['appSecret'],
+        JSON: true
+    };
+    makeApiCall(oauthOptions, makeVideoArray);  
+}
+
+function makeVideoObject(data){
+    var object = {};
+    var dateOptions = { year: 'numeric', month: '2-digit', day: '2-digit', 
+                        hour: '2-digit', minute: '2-digit', second: '2-digit' };
+    object['_id'] = 'fb' + data.id;
+    date = new Date(data.created_time);
+    object.date = date.toLocaleString('ko-KR', dateOptions);
+    object.from = data.from;
+    object.text = data.description;
+    object.videoUrl = data.source;
+    object.likes = data.likes.summary.total_count;
+    object.comments = data.comments.summary.total_count;
+    object.type = 'video';
+    
+    return object;
```

Comment:
var dateOptions = {/** the options above */};
var date = new Date(data.created_time);
return {
    _id: 'fb' + data.id,
    date: date.toLocaleString('ko-KR', dateOptions),
    from: data.from
    /** and so on */
}